addons/sourcemod/scripting/FamilyShareManager.sp

```diff
@@ -480,33 +419,42 @@ stock int GetClientOfAuthId(int authid)
 
 public void SteamWorks_OnValidateClient(int ownerauthid, int authid)
 {
-    if (GetConVarInt(g_hCvar_Method) != 1)
+    if (g_iReject < 1)
         return;
 
     int client = GetClientOfAuthId(authid);
 
+    if (client < 1 || client > MaxClients)
+        return;
+
+    if (IsFakeClient(client) || IsClientSourceTV(client))
+        return;
+
     if (CheckWhiteList(client))
         return;
 
-    if(ownerauthid != authid)
+    if (ownerauthid != authid)
     {
-        LogMessage("Kicking %L (Family share)", client);
-        char rejectMessage[255]; GetConVarString(g_hCvar_RejectMessage, rejectMessage, sizeof(rejectMessage));
-        KickClient(client, rejectMessage);
+        ApplyPunishement(client);
     }
+}
 
-    /*
-    //Now using SteamWorks:
-    EUserHasLicenseForAppResult result = SteamWorks_HasLicenseForApp(client, g_hCvar_AppId.IntValue);
-
-    //Debug text: PrintToServer("Client %N License Value: %i", client, view_as<int>(result));
+stock void ApplyPunishement(int client)
+{
+    char rejectMessage[255];
+    GetConVarString(g_hCvar_RejectMessage, rejectMessage, sizeof(rejectMessage));
 
-    //No License, kick em:
-    if(result > k_EUserHasLicenseResultHasLicense)
+    switch (g_iReject)
     {
-        char rejectMessage[255]; GetConVarString(g_hCvar_RejectMessage, rejectMessage, sizeof(rejectMessage));
-        KickClient(client, rejectMessage);
+        case (2):
+        {
+            LogAction(-1, -1, "Banning %L for %d minutes (Family share)", client, g_iRejectDuration);
+            ServerCommand("sm_ban #%i %d \"%s\"", GetClientUserId(client), g_iRejectDuration, rejectMessage);

```

Comment:
prefer to use built in functions instead of ServerCommand

Reply:
Yeah, I agree but does that mean we need to add support for sbpp? For me sm_ban should be the only cmd without usage of built-in cmd to avoid creating 3rd support

Reply:
i know its a bit more work but its a better approach